### ISSUE TITLE
[ffmpeg] support community builds that disable unneeded codecs/filters/(de)muxers/devices/...

### ISSUE THREAD

**Issue by mcmtroffaes** [microsoft/vcpkg#17237]
****

**Is your feature request related to a problem? Please describe.**
It is very common for users to wish to do custom FFmpeg builds that disable unneeded features, to save resources such as build times or file sizes. FFmpeg supports hundreds of configure options to enable or disable specific filters, codecs, muxers, demuxers, devices, ... It's not feasible for vcpkg to expose each individual feature of FFmpeg. It would be great if vcpkg were to allow community builds to disable unneeded features. The question is whether this is feasible, and if so, how best to achieve this.

**Proposed solution**
One solution would be to adding a text file to the FFmpeg port folder where users can add their own configure options. These options would then be read and passed to the configure script by the FFmpeg port file.

**Describe alternatives you've considered**
I'm not sure. I'm posting this here to elicit some feedback as to whether the proposed solution is a sensible approach, or whether some other/better solution for this already exists in vcpkg.

**Additional context**
N/A

**Comment by Neumann-A** [comment]
Expose all features in the FFMPEG port. Define common features sets as separate features. Let people not covered by this case use manifest mode and explicitly list the features? I don't see a problem in exposing a lot of feature flags as long as they are additive.

**Comment by mcmtroffaes** [comment]
Yes, it would be great to have a single feature for each individual codec, filter, (de)muxer, etc. but wouldn't that be too bloated? Can vcpkg handle several hundreds of features?

**Comment by Neumann-A** [comment]
> Yes, it would be great to have a single feature for each individual codec, filter, (de)muxer, etc. but wouldn't that be too bloated? 

Only export features which can actually be toggled and make sense for a user to toggle. E.g. qt(6)base has >400 possible features but some of them don't make sense, e.g. `FEATURE_xml` can be toggled but the build will simply fail because the targets behind that feature are used elsewhere in the buildsystem.

> Can vcpkg handle several hundreds of features?

Why shouldn't it be able to handle it?

**Comment by mcmtroffaes** [comment]
> Only export features which can actually be toggled and make sense for a user to toggle. E.g. qt(6)base has >400 possible features [...]

It's encouraging to know that vcpkg can handle packages with over 400 individual features.

> > Can vcpkg handle several hundreds of features?
> 
> Why shouldn't it be able to handle it?

I take this as a yes, then. I've simply not seen packages with so many features in vcpkg. I'm interested in learning what's the best approach for the huge set of ffmpeg options in vcpkg, and whether there's an interest in exposing these in some way.

**Comment by pk5ls20** [comment]
So four years have passed—may I politely enquire whether there has been any progress?

**Comment by mcmtroffaes** [comment]
This feature was implemented and then discussed here: https://github.com/microsoft/vcpkg/pull/17323 I don't think anyone has tried to push this forward since.
